ethereum/contracts/BasicOutboundChannel.sol

```diff
@@ -6,14 +6,19 @@ import "./OutboundChannel.sol";
 
 // BasicOutboundChannel is a basic channel that just sends messages with a nonce.
 contract BasicOutboundChannel is OutboundChannel {
-    constructor() {
-        nonce = 0;
-    }
+
+    uint64 public nonce;
+
+    event Message(
+        address source,
+        uint64 nonce,
+        bytes payload
+    );
 
     /**
      * @dev Sends a message across the channel
      */
-    function submit(bytes memory payload) public override {
+    function submit(address, bytes calldata payload) external override {
```

Comment:
havent seen this syntax, with unnamed parameter. what's it do? is the parameter optional?

Reply:
The parameter is still mandatory, and needs to be passed to submit. However in this specific implementation of submit, the parameter doesn't need a name because its not being used. Its the only way to workaround the "unused parameter" warnings emitted by the compiler